service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/PertinentInformationObservationValueMapper.java

```diff
@@ -38,6 +42,11 @@
             .put(DateTimeType.class, value -> processDateTimeType((DateTimeType) value))
             .put(Period.class, value -> processPeriod((Period) value))
             .build();
+    private static final Map<Class<? extends IBaseElement>, Function<IBaseElement, String>> COMPONENT_VALUE_FUNCTIONS =
+        ImmutableMap.<Class<? extends IBaseElement>, Function<IBaseElement, String>>builder()
+            .put(Quantity.class, value -> processComponentValueQuantity((Quantity) value))
+            .put(StringType.class, value -> processComponentValueString((StringType) value))
+            .build();
```

Comment:
`Map.of()` is a lot simpler, imo

```suggestion
    private static final Map<Class<? extends IBaseElement>, Function<IBaseElement, String>> COMPONENT_VALUE_FUNCTIONS = Map.of(
            Quantity.class, value -> processComponentValueQuantity((Quantity) value),
            StringType.class, value -> processComponentValueString((StringType) value)
    );
```